### PR TITLE
split up of lead column structure and field data fixtures

### PR DESCRIPTION
**Pull request by mollux**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

When testing and working on the PHP 8 support (see https://github.com/mautic/mautic/pull/9969#issuecomment-1020322075), I noticed there are some fixtures that contain [DDL queries](https://www.geeksforgeeks.org/sql-ddl-dql-dml-dcl-tcl-commands/).

As Doctrine processes fixtures as 1 transaction that can be rolled back, and DDL queries force a commit when executed,
this results in an error when Doctrine tries to commit the transaction again.

in PHP < 8, this error was silently ignored, but in PHP >= 8 this error results in a ` There is no active transaction` PDOException.

One of the fixtures adds the optional field data to the lead_fields table (OK), and creates the columns for the optional leads and company fields (NOT OK)

This PR splits up those 2 tasks.
The column creation is moved to an eventlistener (`postGenerateSchema`)

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. check if all the expected optional contact fields (e.g. facebook, fax, ... ) are present, both on a contact and on the admin page where you can disable them

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11059"></a>

### CODE REVIEW

File: app/bundles/InstallBundle/Config/config.php
Comment (escopecz) [inline app/bundles/InstallBundle/Config/config.php]:
```suggestion
                'class'     => \Mautic\InstallBundle\EventListener\DoctrineEventSubscriber::class,
```

**Review by escopecz — DISMISSED** [review]
The code looks good. One small suggestion but approving with or without it. Thanks! 👍

**Review by kuzmany — APPROVED** [review]
Nice solution. 
It works as expected  👍

**Comment by RCheesley** [comment]
Great, thanks for the work on getting to the bottom of this @mollux - LGTM! :rocket:

**Pull request merged**
